clojure-mode.el

```diff
@@ -3067,12 +3070,19 @@ With universal argument \\[universal-argument], act on the \"top-level\" form."
 
 \\{clojurec-mode-map}")
 
+;;;###autoload
+(define-derived-mode clojuredart-mode clojure-mode "ClojureDart"
+  "Major mode for editing ClojureDart code.
```

Comment:
Not sure if we need/want a new major mode currently, note that e.g. Babashka doesn't get one.

babashka works in clojure-mode which makes sense and has no reported issues.

Likewise `clojurec-mode` works with arbitrary reader conditionals e.g. `:cljs` or `:bb`

Reply:
Babashka, however, is supposed to be closer to Clojure. Generally the derived modes make more sense for languages with a different runtime and syntactic oddities. I'm not familiar with ClojureDart, so it's hard to say something concrete about it, though.

Reply:
[These are the differences](https://github.com/Tensegritics/ClojureDart/blob/main/doc/differences.md)

Reply:
From what I can see in https://github.com/clojure-emacs/clojure-mode/issues/615#issuecomment-1101692937 , none of these differences so far warrants a distinct mode - they don't seem to imply changes to the things clojure-mode is concerned with e.g. syntax coloring.

It's worth noting that other target runtimes such as CLR, Erlang, etc haven't gotten a mode either.

"No is temporary, yes is forever", so we could just keep it simpler and grow it from there?

Reply:
In practice it doesn't really matter much. The derived modes are very "cheap" and light, so it's not like they add much complexity. I added a derived mode for ClojureScript and clojurec mostly to simplify some checks in CIDER (it's easier to check the major mode than file extensions) and to get cleaner buffer grouping. As runtime like CLR and Erlang never took off there wasn't much demand to do anything about them.

Reply:
Certainly! it makes totally sense. Reverted the commit to remove clojuredart-major-mode, and just added cljd extension to auto-mode-alist clojure-mode.